storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -390,53 +444,98 @@
       [(.getNodeId slot) (.getPort slot)]
       )))
 
+(defn- get-version-for-key [key nimbus-host-port-info conf]
+  (let [version (KeyVersion. key nimbus-host-port-info)]
+    (.getKeyVersion version conf)))
+
+(defn get-key-seq-from-blob-store [blob-store]
+  (let [key-iter (.listKeys blob-store nimbus-subject)]
+    (iterator-seq key-iter)))
+
 (defn- setup-storm-code [nimbus conf storm-id tmp-jar-location storm-conf topology]
-  (let [stormroot (master-stormdist-root conf storm-id)]
-   (log-message "nimbus file location:" stormroot)
-   (FileUtils/forceMkdir (File. stormroot))
-   (FileUtils/cleanDirectory (File. stormroot))
-   (setup-jar conf tmp-jar-location stormroot)
-   (FileUtils/writeByteArrayToFile (File. (master-stormcode-path stormroot)) (Utils/serialize topology))
-   (FileUtils/writeByteArrayToFile (File. (master-stormconf-path stormroot)) (Utils/toCompressedJsonConf storm-conf))
-   (if (:code-distributor nimbus) (.upload (:code-distributor nimbus) stormroot storm-id))
-   ))
+  (let [subject (get-subject)
+        storm-cluster-state (:storm-cluster-state nimbus)
+        blob-store (:blob-store nimbus)
+        jar-key (master-stormjar-key storm-id)
+        code-key (master-stormcode-key storm-id)
+        conf-key (master-stormconf-key storm-id)
+        nimbus-host-port-info (:nimbus-host-port-info nimbus)]
+    (when tmp-jar-location ;;in local mode there is no jar
+      (.createBlob blob-store jar-key (FileInputStream. tmp-jar-location) (SettableBlobMeta. BlobStoreAclHandler/DEFAULT) subject)
+      (if (instance? LocalFsBlobStore blob-store)
+        (.setup-blobstore! storm-cluster-state jar-key nimbus-host-port-info (get-version-for-key jar-key nimbus-host-port-info conf))))
+    (.createBlob blob-store conf-key (Utils/toCompressedJsonConf storm-conf) (SettableBlobMeta. BlobStoreAclHandler/DEFAULT) subject)
+    (if (instance? LocalFsBlobStore blob-store)
+      (.setup-blobstore! storm-cluster-state conf-key nimbus-host-port-info (get-version-for-key conf-key nimbus-host-port-info conf)))
+    (.createBlob blob-store code-key (Utils/serialize topology) (SettableBlobMeta. BlobStoreAclHandler/DEFAULT) subject)
+    (if (instance? LocalFsBlobStore blob-store)
+      (.setup-blobstore! storm-cluster-state code-key nimbus-host-port-info (get-version-for-key code-key nimbus-host-port-info conf)))))
+
+(defn- read-storm-topology [storm-id blob-store]
+  (Utils/deserialize
+    (.readBlob blob-store (master-stormcode-key storm-id) (get-subject)) StormTopology))
+
+(defn- get-blob-replication-count [blob-key nimbus]
+  (if (:blob-store nimbus)
+          (-> (:blob-store nimbus)
+            (.getBlobReplication  blob-key nimbus-subject))))
```

Comment:
Fix indentation here. The threading macro should be indented two spaces in from the `if` on the previous line, and the call to `getBlobReplication` should be indented the same amount as `(:blob-store nimbus)`.


Also, here and elsewhere, when we are adding or changing clojure functions, we need to put the argument list on its own line in order to follow idiomatic clojure style.  There was a change we merged in a while ago that fixed many of these instances, and we should try to maintain the same style going forward.